### PR TITLE
add `deprecated_inequality_operator` pattern

### LINKED ISSUE #49: Replace deprecated `<>` operator with `!=`
Related to #21. 

I'll create an issue as I go, to make it more visible which ones I are finished.

Ref: [Avoid deprecated <> operator](https://www.quantifiedcode.com/knowledge-base/maintainability/Avoid%20deprecated%20%60%3C%3E%60%20operator/348m67d2)

### PR DESCRIPTION
**Pull request by lwm**
Closes #49.

Ran out of time, will be back. I'll leave this here for now ;)

### CODE REVIEW

File: undebt/examples/deprecated_inequality_operator.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+from __future__ import division
+from __future__ import print_function
+
+from pyparsing import Word
+
+from undebt.pattern.python import EXPR
+from undebt.pattern.util import tokens_as_list
+
+
+grammar = EXPR + Word("<>") + EXPR
```
Comment (ajm188) [inline undebt/examples/deprecated_inequality_operator.py]: You can add `.suppress()` onto the `<>` token, then your token list will only contain the two expressions.

File: undebt/examples/deprecated_inequality_operator.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+
+@tokens_as_list(assert_len=2)
```
Comment (ajm188) [inline undebt/examples/deprecated_inequality_operator.py]: Your pattern actually grabs 3 tokens, so this assertion will fail. See my above comment about suppressing, or change the assertion.

File: undebt/examples/deprecated_inequality_operator.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+def replace(tokens):
+    """foo <> bar -> foo != bar"""
+    return tokens[0] + "!=" + tokens[2]
```
Comment (ajm188) [inline undebt/examples/deprecated_inequality_operator.py]: If you suppress the `<>` operator, the second token is actually `tokens[1]`

File: undebt/examples/deprecated_inequality_operator.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+
+extra = "from function_lives_here import function"
```
Comment (ajm188) [inline undebt/examples/deprecated_inequality_operator.py]: I'm assuming this is accidental copy-pasta leftovers?

Comment (lwm) [inline undebt/examples/deprecated_inequality_operator.py]: Yep, will remove!

File: undebt/examples/deprecated_inequality_operator.py (hunk 1 above)
Comment (lwm) [inline undebt/examples/deprecated_inequality_operator.py]: Oh nice, will add that.

Comment (evhub) [inline undebt/examples/deprecated_inequality_operator.py]: This should be `Literal` not `Word`.

Comment (ajm188) [inline undebt/examples/deprecated_inequality_operator.py]: Good catch

**Comment by ajm188** [comment]
@lwm FYI: #52 is going to break your change if it gets merged first. Are you on the gitter? We can coordinate how we want to proceed.

**Comment by ajm188** [comment]
@lwm merged. Rebase away

**Comment by lwm** [comment]
@ajm188 Done :+1:

File: undebt/examples/deprecated_inequality_operator.py [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+from __future__ import division
+from __future__ import print_function
+
+from pyparsing import Word
+
+from undebt.pattern.lang.python import UNARY_OP_ATOM
+from undebt.pattern.util import tokens_as_list
+
+
+grammar = UNARY_OP_ATOM + Word("<>").suppress() + UNARY_OP_ATOM
```
Comment (ajm188) [inline undebt/examples/deprecated_inequality_operator.py]: Why did you change `EXPR` to `UNARY_OP_ATOM`?

Comment (ajm188) [inline undebt/examples/deprecated_inequality_operator.py]: And as @evhub said prior, you want the `<>` operator to be a `Literal`, not a `Word`.

Comment (evhub) [inline undebt/examples/deprecated_inequality_operator.py]: `UNARY_OP_ATOM` is actually necessary, since `EXPR` will actually match `<>` as a valid operator.

Comment (ajm188) [inline undebt/examples/deprecated_inequality_operator.py]: Ah okay

File: undebt/examples/deprecated_inequality_operator.py (hunk 1 above)
Comment (lwm) [inline undebt/examples/deprecated_inequality_operator.py]: Ah yes, just checked the docs now. Fixed.

**Comment by ajm188** [comment]
lg2m! merging
